### PR TITLE
google_os_config_os_policy_assignment

### PR DESCRIPTION
**Pull request by trodge**
Upgraded DCL and added overrides for osconfig os policy assignment.
Reworked AppendToBasePath override into BasePathReplacement.
Implemented serialization for map[string]string types.

fixes https://github.com/hashicorp/terraform-provider-google/issues/10628

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_os_config_os_policy_assignment`
```

### CODE REVIEW

File: tpgtools/overrides/monitoring/beta/monitored_project.yaml [hunk 1]
```diff
@@ -12,6 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-- type: APPEND_TO_BASE_PATH
+- type: BASE_PATH_REPLACEMENT
   details:
-    string: v1
+    string: 'client.Config.BasePath + "v1"'
```
Comment (ndmckinley) [inline tpgtools/overrides/monitoring/beta/monitored_project.yaml:17]: I don't think we want to be putting code in these files.  I know it's not _much_ code, but it's really worth avoiding in my opinion.  Is that feasible?

**Review by ndmckinley — COMMENTED** [review]
This looks fine to me with the exception of the design of the trait.  Riley or Sam could overrule me on that point though - I don't feel super strongly about it.

File: tpgtools/overrides/monitoring/beta/monitored_project.yaml (hunk 1 above)
Comment (trodge) [reply tpgtools/overrides/monitoring/beta/monitored_project.yaml:17]: It's feasible if we want to keep BasePathReplacement and AppendToBasePath as two separate traits. To me it seems cleaner to combine them into one, even if it means we have to do it like this.

File: tpgtools/overrides/osconfig/beta/os_policy_assignment.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- type: REPLACE_IN_BASE_PATH
+  details:
+    old: v1beta
+    new: v1
```
Comment (ndmckinley) [inline tpgtools/overrides/osconfig/beta/os_policy_assignment.yaml:18]: Thank you!  Yes, I think this is much better.

**Review by ndmckinley — APPROVED** [review]

**Review by rileykarson — COMMENTED** [review]
I haven't reviewed the PR here, I only took a look at the generated code. I'm seeing the `v1` part drop from monitored project in the downstream, and I'm not sure I can see where that's getting added back somewhere else: [modular-magician/terraform-provider-google-beta@`auto-pr-5481-old..auto-pr-5481`#diff-4453ffdcd0](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5481-old..auto-pr-5481#diff-4453ffdcd0c88d5bbf8e263267b9c0b6ddf437c2c4d39555c4ecaac20f8cd1dc)L94

**Comment by rileykarson** [comment]
I think that's supported by the VCR failure, where we're making a different request than we did before:

```
------- Stdout: -------
=== RUN   TestAccMonitoringMonitoredProject_BasicMonitoredProject
=== PAUSE TestAccMonitoringMonitoredProject_BasicMonitoredProject
=== CONT  TestAccMonitoringMonitoredProject_BasicMonitoredProject
provider_test.go:278: Step 1/2 error: Error running apply: exit status 1
Error: Error creating MonitoredProject: failed to create a diff: failed to retrieve MonitoredProject resource: Get "https://monitoring.googleapis.com/locations/global/metricsScopes/ci-test-project-188019?alt=json": Requested interaction not found
on terraform_plugin_test.tf line 2, in resource "google_monitoring_monitored_project" "primary":
2: resource "google_monitoring_monitored_project" "primary" {
--- FAIL: TestAccMonitoringMonitoredProject_BasicMonitoredProject (24.90s)
FAIL
```

**Comment by ndmckinley** [comment]
Yep - you're right about that.  I'll rereview when the tests are green.

**Comment by ndmckinley** [comment]
For some reason, it doesn't seem like re-running the failed VCR tests is running for you.  I can't explain that, sorry!  Hm, can you maybe run the failed tests on your machine locally?  I don't suspect any of them are related, but you never know...

**Comment by ndmckinley** [comment]
Those are all problems with billing enablement and an internal API error.  No need to change anything on those.  Works fine for me.

Should we worry about the TF validator?

**Review by ndmckinley — APPROVED** [review]

**Comment by ndmckinley** [comment]
nope, don't need to worry about that one, known issue seems like.

**Pull request merged**